### PR TITLE
Replay demo

### PR DESCRIPTION
**Pull request by bergundy**

### CODE REVIEW

File: README.md
Comment (bergundy) [inline README.md:1]: This whole README should be redone to explain what's in this repo and how to use it.

File: scripts/create-certs-dir.js
Comment (bergundy) [inline scripts/create-certs-dir.js:1]: Can just be a shell script embedded into ci.yml but this works on windows too if users want to test it out.

**Review by bergundy — COMMENTED** [review]
Note the new workflow versions are in different branches now.

File: scripts/create-certs-dir.js
Comment (Sushisource) [reply scripts/create-certs-dir.js:1]: This is better anyway I think. Embedding them directly into CI is mostly just a pure convenience thing but is pretty much never really better.

File: src/env.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    return { workflowBundle: { codePath: workflowBundlePath } };
+  } else {
+    return { workflowsPath: require.resolve("./workflows") };
+  }
+}
+
+export const namespace = getenv("TEMPORAL_NAMESPACE", "default");
+// TODO: change for actual demo (avoids clutter created during tests)
+// export const taskQueue = "versioning-demo";
+export const taskQueue = "versioned-queue";
```
Comment (Sushisource) [inline src/env.ts:52]: You might want to change this now and just delete any existing workflows beforehand. Maybe easier than remembering to change it.

File: src/check-compatibility.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  namespace,
+  taskQueue,
+} from "./env";
+
+async function run() {
+  const prTitle = process.argv[2];
+  if (!prTitle) {
+    throw new Error(`usage: ${process.argv[1]} <prTitle>`);
+  }
+  const matches = prTitle.match(/^\[compatible-([^\]]+)\]/);
```
Comment (Sushisource) [inline src/check-compatibility.ts:15]: nit: You'd probably not want `[` too

File: src/check-compatibility.ts [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    .list({
+      query: `TaskQueue="${taskQueue}" AND BuildIds IN (${joinedBuildIds})`,
+    })
+    .intoHistories();
+  const replayResults = Worker.runReplayHistories(
+    getWorkflowOptions(),
+    histories
+  );
+  let i = 0;
+  for await (const result of replayResults) {
```
Comment (Sushisource) [inline src/check-compatibility.ts:49]: Not that it's hugely important, but it's maybe slightly performative to do this loop after we actually ran the histories.

Maybe actually important, the limiter doesn't really limit anything meaningful.

It'd be best to make a basic wrapper for the async iterator that does this so it actually happens during `runReplayHistories`

**Review by Sushisource — APPROVED** [review]

File: src/check-compatibility.ts (hunk 3 above)
Comment (Sushisource) [reply src/check-compatibility.ts:49]: Oh wait nevermind I forgot the output is also an async iterator

File: src/check-compatibility.ts (hunk 2 above)
Comment (bergundy) [reply src/check-compatibility.ts:15]: I think it's good enough, there might be other characters that we don't want but I'll leave it as an exercise for the reader.

File: src/env.ts (hunk 1 above)
Comment (bergundy) [reply src/env.ts:52]: I'll do it after we do a dry run.

**Pull request merged**